test/TestProject/UnitTest/InventoryRepositorytest.cs

```diff
@@ -0,0 +1,160 @@
+    public void ShouldAddTwoInventoryItem()
+    {
+        //Given
+        Inventory? inventory = InventoryFixture.BuildInventory();
+        InventoryItem firstItem = new InventoryItem { ID = 0, product = new Product { ID = 1, Name = "test", Status = 'A' }, ItemCount = 100, Price = 0, Status = 'A' };
+        InventoryItem secondItem = new InventoryItem { ID = 0, product = new Product { ID = 2, Name = "test2", Status = 'A' }, ItemCount = 200, Price = 0, Status = 'A' };
+
+        //When
+        _sut.Add(inventory);
+        _sut.AddItem(1, firstItem);
+        _sut.AddItem(1, secondItem);
+        var inv = _sut.GetByID(1);
+
+        //Then
+        Assert.Equal(3, inv?.items?.Count());
+    }
+
+    [Fact]
+    public void ShouldDeleteOneInventoryItem()
+    {
+        //Given
+        Inventory? inventory = InventoryFixture.BuildInventory();
+        InventoryItem firstItem = new InventoryItem { ID = 0, product = new Product { ID = 1, Name = "test", Status = 'A' }, ItemCount = 100, Price = 0, Status = 'A' };
+        InventoryItem secondItem = new InventoryItem { ID = 0, product = new Product { ID = 2, Name = "test2", Status = 'A' }, ItemCount = 200, Price = 0, Status = 'A' };
+
+        //When
+        _sut.Add(inventory);
+        _sut.AddItem(1, firstItem);
+        _sut.AddItem(1, secondItem);
+        var inv = _sut.GetByID(1);
+        Assert.Equal(3, inv?.items?.Count());
+
+        _sut.DeleteItem(1, 3);
+        var (invet, item) = _sut.GetItem(1, 2);
+        //Then
+        Assert.Equal(2, _sut.GetByID(1)?.items?.Count());
+        Assert.Equal("test", item?.product?.Name);
+        Assert.Equal(0, invet.items?.Where(i => i.product?.Name == "DeleteMe").Count());
+    }
+
+    [Fact]
+    public void ShouldDeleteOneInventory()
+    {
+        //Given
+        List<Inventory> inventories = InventoryFixture.BuildInventory(3);
+
+        inventories.ForEach(inv =>
+        {
+            _sut.Add(inv);
+        });
+
+        //When
+        _sut.Delete(1);
+        var remainingInventory = _sut.GetByID(2);
+        var inventoryNotFound = _sut.GetByID(1);
+
+        //Then
+        Assert.Equal(2, _sut.GetAll().Count);
+        Assert.Equal(2, remainingInventory?.ID);
+        Assert.Null(inventoryNotFound);
+    }
+
+    [Fact]
+    public void ShouldUpdateTheInventory()
+    {
+        //Given
+        Inventory? inventory = InventoryFixture.BuildInventory();
+        Assert.NotNull(inventory);
+        Assert.Equal(1, inventory.ID);
+
+        _sut.Add(inventory);
+
+        Inventory newInventory = inventory with { ToTalUnits = 200 };
+        Assert.Equal(200, newInventory.ToTalUnits);
+
+        //When
+        _sut.Update(newInventory);
+
+        //Then
+        Inventory? updatedInventory = _sut.GetByID(1);
+        Assert.Equal(200, updatedInventory?.ToTalUnits);
+    }
+
+    [Fact]
+    public void ShouldUpdateOneInventoryItem()
+    {
+        //Given
+        Inventory? inventory = InventoryFixture.BuildInventory();
```

Comment:
Why a question mark here?

Reply:
The `?` will avoid the program to break if the method `BuildInventory()` return null. In other words, will make the variable nullable.

Reply:
Why could it be null if you are telling the function to build an inventory for you?

Reply:
This is a check added in this version of .NET Core. As you mention, the method will never return `null` but this is a global check. I put it there to satisfy the compiler that will give me a warning in those cases. So, can be removed. Should I do it?

Reply:
This is really weird if the compiler thinks this can be null. Not pretty sure if this a expected behavior.

Reply:
it is weird in some cases. Right now in this case I can remove it and the program will compile but I will have a warning of potentially null. Since the compiler is looking that this variable is getting something from a method, is not actually verifying if method returns null or something, just warn of a potentially null. Since .NET Core 5, this is a default behavior activated in the `.csproj` file.